lib/runtime/wazero/imports.go

```diff
@@ -0,0 +1,2247 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package wazero_runtime
+
+import (
+	"context"
+	"crypto/rand"
+	"encoding/binary"
+	"fmt"
+	"math/big"
+	"reflect"
+	"time"
+
+	"github.com/ChainSafe/gossamer/internal/log"
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto"
+	"github.com/ChainSafe/gossamer/lib/crypto/ed25519"
+	"github.com/ChainSafe/gossamer/lib/crypto/secp256k1"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/lib/runtime"
+	"github.com/ChainSafe/gossamer/lib/transaction"
+	"github.com/ChainSafe/gossamer/lib/trie"
+	"github.com/ChainSafe/gossamer/lib/trie/proof"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+	"github.com/tetratelabs/wazero/api"
+)
+
+var (
+	logger = log.NewFromGlobal(
+		log.AddContext("pkg", "runtime"),
+		log.AddContext("module", "wazero"),
+	)
+)
+
+const (
+	validateSignatureFail = "failed to validate signature"
+)
+
+// toPointerSize converts an uint32 pointer and uint32 size
+// to an int64 pointer size.
+func newPointerSize(ptr, size uint32) (pointerSize uint64) {
```

Comment:
This is not properly casting to int64. If this is still the needed functionality should be 
`return int64(ptr) | (int64(size) << 32)`

Reply:
ah okay its the comment that wasnt updated